src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -277,8 +274,10 @@ internal partial class CodeFixService : ICodeFixService
                 FixAllCodeActionContext fixAllContext = null;
                 if (fixAllProviderInfo != null)
                 {
-                    var codeFixProvider = (fixer as CodeFixProvider) ?? new WrapperCodeFixProvider((ISuppressionFixProvider)fixer, diagnostics);
-                    fixAllContext = FixAllCodeActionContext.Create(document, fixAllProviderInfo, codeFixProvider, diagnostics, this.GetDocumentDiagnosticsAsync, this.GetProjectDiagnosticsAsync, cancellationToken);
+                    var codeFixProvider = (fixer as CodeFixProvider) ?? new WrapperCodeFixProvider((ISuppressionFixProvider)fixer, diagnostics.Select(d => d.Id).Distinct().ToImmutableArray());
```

Comment:
to break dependency to Diagnostics from WrapperCodeFixProvider. only reason it wanted to have Diagnostic was to get ID list. because of it all caller must have Diagnostics making caller that only has DiagnosticData to be required to convert them to Diagnostic first and then get ID and then throw Diagnostics away.